FluidNC/src/Kinematics/Skewed.cpp

```diff
@@ -0,0 +1,132 @@
+#include "Skewed.h"
+
+#include "src/Machine/MachineConfig.h"
+#include "src/Machine/Axes.h"  // ambiguousLimit()
+#include "src/Limits.h"
+
+namespace Kinematics {
+
+    void SkewAxis::group(Configuration::HandlerBase& handler) {
+        handler.item("dist", _dist, 1.0f, 100000.0f );
+        handler.item("x", _x[0], -1000.0f, 1000.0f );
+        handler.item("y", _x[1], -1000.0f, 1000.0f );
+        handler.item("z", _x[2], -1000.0f, 1000.0f );
+        handler.item("a", _x[3], -1000.0f, 1000.0f );
+        handler.item("b", _x[4], -1000.0f, 1000.0f );
+        handler.item("c", _x[5], -1000.0f, 1000.0f );
+    }
+
+    void SkewAxis::init() {
+        log_info( "      Skew ( " << _x[0] << ", " << _x[1] << ", " << _x[2] << " ) over " << _dist << "mm" );
+    }
+
+
+    // This return a row of transformation matrix 
+    void SkewAxis::getRow( const uint count, float* buf ) {
+        for( uint i = 0; i < count; ++i )
+            buf[ i ] = _x[ i ] / _dist + (( i == _axisIdx ) ? 1.0f : 0.0 );
+    }
+
+    ////////////////////
+
+    Skewed::Skewed() : _numberAxis( MAX_N_AXIS ), _axis() {
+        for (int i = 0; i < MAX_N_AXIS; ++i) {
+            _axis[i] = nullptr;
+        }
+
+    }
+
+    void Skewed::init() {
+        GenericCartesian::init();
+
+        bool fail = false;
+
+        if (_mtx)
+            delete _mtx;
+
+        if (_rev)
+            delete _rev;
+
+        _mtx = new Mtx<float>(_numberAxis, _numberAxis);
+        _rev = new Mtx<float>(_numberAxis, _numberAxis);
+
+        for (size_t axis = 0; axis < _numberAxis; axis++) {
+            auto a = _axis[axis];
+            if (a) {
+                a->init();
+                a->getRow( _numberAxis, _mtx->ptr( axis, 0 ) );
```

Comment:
getRow is hard to understand because the SkewAxis method reaches out and injects something into the outside matrix.  Something like this would be clearer and simpler:
```c++
    if (a) {
        a->init();
        auto row = _mtx->ptr(axis, 0);
        for (size_t i = 0; i < _numberAxis; ++i) {
            row[i] = (i == axis) ? 1.0 : 0.0;  // Start with 1 on the diagonals
            row[i] += a->_offsets[i] / a->_axis_length;  // Add adjustment factors
        }
```
With that formulation, _axisIdx and getRow are unnecessary

Reply:
The function returns basis-vector. And return it into provided buffer is fairly typical. Especially on the systems where you don't have defined float2, float3, ... types. 
I would say that when you have one function which build whole matrix is harder to read, but emmm. Ok. I can throw everything into one place.